src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngineUnitTest.java

```diff
@@ -21,12 +23,69 @@
+
+        final Allele REF = Allele.create("A", true);
+        final Allele ALT = Allele.create("T");
+        final List<Allele> alleles = Arrays.asList(new Allele[]{REF, ALT});
+
+
+        final int[] MQs = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, QualityUtils.MAPPING_QUALITY_UNAVAILABLE};
+        final int[] MQ2s = {1, 2, 3, 4, 5, 6, 7, 8, 9, QualityUtils.MAPPING_QUALITY_UNAVAILABLE};
+        final List<Integer> MQsList = Arrays.asList(ArrayUtils.toObject(MQs));
+
+
+        final List<GATKRead> altReads = Arrays.stream(MQs).mapToObj(mq -> ArtificialAnnotationUtils.makeRead(30, mq)).collect(Collectors.toList());
+        final List<GATKRead> refReads = Arrays.stream(MQ2s).mapToObj(mq -> ArtificialAnnotationUtils.makeRead(30, mq)).collect(Collectors.toList());
+
+        final ReadLikelihoods<Allele> likelihoods =
+                ArtificialAnnotationUtils.makeLikelihoods("sample1", refReads, altReads, -10.0, -9.0, REF, ALT);
+
+        final VariantContext vc = ArtificialAnnotationUtils.makeVC();
+        final ReducibleAnnotationData<Object> myData = new ReducibleAnnotationData<>("-10.0");
+
+        new AS_RMSMappingQuality().calculateRawData(vc, likelihoods, myData);
+        Map<String, List<Object>> testAnnotationData = new HashMap<>();
+        testAnnotationData.put(new AS_RMSMappingQuality().getRawKeyName(), Collections.singletonList(myData));
+
+        Map<String, Object> value = vae.combineAnnotations(alleles, testAnnotationData);
+        Assert.assertEquals(value.get(GATKVCFConstants.AS_RAW_RMS_MAPPING_QUALITY_KEY), "285.00|385.00");
+    }
+
+    @Test
+    public void testFinalizeAnnotations() throws Exception {
+        final List<String> annotationGroupsToUse = Collections.emptyList();
+        final List<String> annotationsToUse = Collections.singletonList(AS_RMSMappingQuality.class.getSimpleName());
+        final List<String> annotationsToExclude = Collections.emptyList();
+        final FeatureInput<VariantContext> dbSNPBinding = null;
+        final List<FeatureInput<VariantContext>> features = Collections.emptyList();
+        final VariantAnnotatorEngine vae = VariantAnnotatorEngine.ofSelectedMinusExcluded(annotationGroupsToUse, annotationsToUse, annotationsToExclude, dbSNPBinding, features);
+        final Allele refAllele = Allele.create("A", true);
+        final Allele altAllele = Allele.create("T");
+        final Genotype genotype = new  GenotypeBuilder("sample2", Arrays.asList(refAllele, altAllele))
+                .AD(new int[]{8,9}).make();
+
+        final VariantContext vc = new VariantContextBuilder(ArtificialAnnotationUtils.makeVC()).attribute(GATKVCFConstants.AS_RAW_RMS_MAPPING_QUALITY_KEY, "285.00|385.00").genotypes(genotype).make();
+        final VariantContext result = vae.finalizeAnnotations(vc, vc);
+        Assert.assertNull(result.getAttribute(GATKVCFConstants.AS_RAW_RMS_MAPPING_QUALITY_KEY));
+        Assert.assertNotNull(result.getAttribute(GATKVCFConstants.AS_RMS_MAPPING_QUALITY_KEY));
```

Comment:
Why aren't you checking the value here?

Reply:
Added it, though this test is more about testing the finalize function rather than the annotation itself.

Reply:
I added an assertion for an unrelated key.